crates/red_knot_python_semantic/src/types.rs

```diff
@@ -344,7 +344,7 @@ bitflags! {
     ///
     /// [1]: https://docs.python.org/3/library/dataclasses.html
     #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-    pub struct DataclassMetadata: u16 {
+    pub struct DataclassParams: u16 {
```

Comment:
New name in analogy to the `__dataclass_params__` runtime introspection attribute, which contains the same information.